subsys/bluetooth/controller/ll_sw/ll_filter.c

```diff
@@ -384,12 +534,18 @@ u32_t ll_rl_add(bt_addr_le_t *id_addr, const u8_t pirk[16],
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
+	i = ll_rl_find(id_addr->type, id_addr->a.val);
```

Comment:
lets have ll_rl_find(type, addr, *free_idx); so that if not found use free_idx to insert device. Saves a subsequest loop to find empty slot.